### ISSUE TITLE
build videos?

### ISSUE THREAD

**Issue by NIXOYE** [Rose-Hulman-ROBO4xx/1314-BeagleBone-Quadcopter#5]
are their any plans on making videos on the process of putting one together?

**Comment by rockybulwinkle** [comment]
Not really.  This was a senior design project.  As far as I am aware, everyone has moved on.  Please note that master_rev2 became our main branch.  At some point I am going to merge it into master.

**Comment by rockybulwinkle** [comment]
All that said, the construction of the frame is fairly simple.  If you go to master_rev2, there are pictures.

**Comment by NIXOYE** [comment]
can you link it? also, was there any work on adding gps or wifi to it so it could follow a device? so you could have it trail behind you or be able to get to a saved location?

**Comment by rockybulwinkle** [comment]
We had planned on a GPS module.  IIRC, it even has a spot where you'd solder it to the PCB.  But we didn't get to it before we graduated.  In the software's current state, it only supports R/C mode.

**Comment by rockybulwinkle** [comment]
By R/C mode, I mean dual analog sticks that control throttle, yaw, roll, and pitch.

**Comment by rockybulwinkle** [comment]
Take a look at the master_rev2 home page:
https://github.com/Rose-Hulman-ROBO4xx/1314-BeagleBone-Quadcopter/tree/master_rev2

**Comment by NIXOYE** [comment]
oh, wow did you guys not see how amazing this project is and just get the grade forget about it? was the beaglebone detachable from the copter parts easily? also is there also open guides for the control or remote or whatever you called the device to drive it?

**Comment by NIXOYE** [comment]
its so beautiful.

**Comment by rockybulwinkle** [comment]
No, we get how cool this was ;)

Frankly, after we graduated we didn't have much time to continue development, though some of us wanted to, myself included.  I may still pick up this project again.  It was my baby, ya know?

it wasn't too hard to separate the bone from the frame.  IIRC, you unscrewed four screws and carefully remove it from the PCB.

**Comment by NIXOYE** [comment]
did you guys try carrying anything with it? i hope you do pick this up again!

**Comment by rockybulwinkle** [comment]
No, we didn't try carrying anything with it.  The controls were a little unwieldily as it was.  And I say that as an R/C pilot with more than 15 years of experience.

**Comment by rockybulwinkle** [comment]
Somehow we had the nerve to fly it in an auditorium full of people XD.

**Comment by rockybulwinkle** [comment]
Be sure to check out the dropbox videos, if you didn't see them already.  I'm the guy with the deeper voice and beard. :p

**Comment by NIXOYE** [comment]
hm, well with gps\cameras itd probably be pretty straightforward to get it to drive itself. it might even be able to use something like openCV to follow directly above you

**Comment by NIXOYE** [comment]
yeah i have the dropbox bookmarked as soon as i came accros this project

**Comment by rockybulwinkle** [comment]
Are you a Rose-Hulman student as well?

**Comment by rockybulwinkle** [comment]
"also is there also open guides for the control or remote or whatever you called the device to drive it?"

To control it, you connect to a web server hosted on the bone (it acts as an access point).  The website uses some joystick API.  We tried using both an xbox controller and a *proper* R/C controller, specifically the DX6.

**Comment by rockybulwinkle** [comment]
commands are sent to control_alg by writing to a FIFO and it constantly writes the latest image from the camera to a ramfs.  So really you could drive it with a program other than the web server if you wish.

**Comment by rockybulwinkle** [comment]
If you have any other questions, just ask.

**Comment by NIXOYE** [comment]
nop, i was looking around for a project i thought i had seen online and came across this. 

it'd be interesting to see if you could set it up with a cameras so it could move around like teslas/geohots comma.ai project and be able to have it avoid people\buildings on its own.

**Comment by NIXOYE** [comment]
how durable were they when they fell. did it actually break that time it fell and someone said it broke?

**Comment by rockybulwinkle** [comment]
They were pretty fragile.  We did move to plywood/bass wood frames, which both helped the weight and improved durability.

**Comment by rockybulwinkle** [comment]
Initially it was acrylic.

**Comment by NIXOYE** [comment]
how many did you all build? did you ever look aronud for what GPS you could add to the soldering spot?

**Comment by rockybulwinkle** [comment]
IIRC, we ordered several of our custom PCBs but only populated maybe 2 or 3.  We built many, many frames worth of parts and backups.

**Comment by rockybulwinkle** [comment]
I forget what GPS we designed it for.  It was a specific model.  I don't think they all share the same footprint, ya know?

**Comment by NIXOYE** [comment]
were there any notes or something kept where it might be? i wonder how small cameras can get now that could work with it without putting to much weight. maybe one on each of the things that copter line up the copter wings.

**Comment by rockybulwinkle** [comment]
Yeah, I'm sure there are notes somewhere.  They might be on my google drive, i.e. not public though.  The camera was an OV7670 which I wrote a PRU program for.  It read the FIFO off the camera and stored it in memory shared with the ARM.  It worked well enough, but was a pain to get going.  It involved a lot of assembly and magic.

IIRC, we went that route with the camera because we wanted to use the single USB port for networking, and didn't want to have a USB hub strapped to the side.
